compiler/next/include/chpl/resolution/scope-types.h

```diff
@@ -82,6 +82,14 @@ class OwnedIdsWithName {
   bool operator!=(const OwnedIdsWithName& other) const {
     return !(*this == other);
   }
+  void mark(Context* context) const {
+    id_.mark(context);
+    if (auto ptr = moreIds_.get()) {
+      for (auto const& elt : *ptr) {
+        context->markOwnedPointer(&elt);
```

Comment:
Even though the elt type isn't owned<ID>, it still counts as owned b/c moreIds_ is owned, is that right? And we consider owned-ness to be transitive?

Reply:
Yes, more or less. It's "owned" here because the type `OwnedIdsWithName` here is responsible for freeing it.